solarforecastarbiter/io/fetch/arm.py

```diff
@@ -174,18 +191,20 @@ def fetch_arm(user_id, api_key, datastreams, start, end):
         requested period.
     """
     site_dfs = []
-    for ds, ds_type in datastreams.items():
-        fns = list_arm_filenames(user_id, api_key, ds, start, end)
-        var_dfs = []
-        for fn in fns:
-            file = retrieve_arm_dataset(user_id, api_key, fn)
-            # TODO: allow users to define ds_type and applicable variables
-            if ds_type == 'irrad':
-                var = extract_arm_variable(file, 'down_short_diffuse_hemisp')
-            elif ds_type == 'weather':
-                var = extract_arm_variable(file, 'temp_mean')
-            var_dfs.append(var)
-            file.close()
-        site_dfs.append(pd.concat(var_dfs, sort=False))
-    new_data = pd.concat(site_dfs, axis=1)
-    return new_data
+    fns = list_arm_filenames(user_id, api_key, datastream, start, end)
+    for fn in fns:
+        nc_file = retrieve_arm_dataset(user_id, api_key, fn)
+        var_dfs = pd.DataFrame()
+        for var in variables:
+            try:
+                var_data = extract_arm_variable(nc_file, var)
+            except IndexError:
+                continue
+            var_dfs = var_dfs.join(var_data, how='outer')
+        nc_file.close()
+        site_dfs.append(var_dfs)
+    if site_dfs:
+        new_data = pd.concat(site_dfs)
```

Comment:
how does this handle repeated variables? Maybe I'm confused about what the columns in the `DataFrame` are expected to be. Might be helpful to add an example to the documentation.

Reply:
The above `var in variables` loop joins the different variable columns from a 1-day file together into a single DataFrame. Then this call concatenates multiple days together, so there shouldn't be any repeated variables. Using `extract_arm_variable` to create a DataFrame from each variable we want from the file here probably isn't efficient. I'll try using a dictionary like you suggested above, that should help to clarify the code as well.